visallo/visallo
# Fix Fit Button On Map

A bug fix and improvement for the "Fit" button on the map product.

Bug fix: When calling `fit`, the center coordinates were being calculated based on the view's old resolution. This was a problem when the new resolution was significantly different than the old one. The code now adjusts the center based on what the resolution will be after the fit.

Improvement: Calculate the extents of the map features based on the individual features rather than the super pins. In addition, update MIN_FIT_ZOOM_RESOLUTION to allow a little closer zoom if the extents don't warrant being zoomed so far out.


Testing Instructions:
1. Add a number of entities and annotations to the map
1. Press fit and ensure that the map resizes properly
1. Open the inspector panel and drag it wide
1. Press fit and ensure that the map resizes properly
1. Remove all entities and annotations except for one
1. Press fit and ensure that the map resizes properly

CHANGELOG
Fixed: A bug in the center calculation when pressing the "Fit" button on the map
